File: HYPERLEDGER/bfn-network-apps/businesslibrary/lib/data/invoice_acceptance.dart
import 'package:businesslibrary/util/Finders.dart';

class InvoiceAcceptance extends Findable {
  String acceptanceId;
  String supplierName;
  String customerName;
  String invoiceNumber;
  String date;
  String invoice;
  String customer, supplier;
  String user;

  InvoiceAcceptance(
      {this.acceptanceId,
      this.supplierName,
      this.customerName,
      this.invoiceNumber,
      this.date,
      this.invoice,
      this.supplier,
      this.customer,
      this.user});

  InvoiceAcceptance.fromJson(Map data) {
    this.acceptanceId = data['acceptanceId'];
    this.invoice = data['invoice'];
    this.customer = data['customer'];
    this.user = data['user'];
    this.date = data['date'];
    this.invoiceNumber = data['invoiceNumber'];
    this.supplier = data['supllier'];
    this.supplierName = data['supplierName'];
    this.customerName = data['customerName'];
  }

  Map<String, dynamic> toJson() => <String, dynamic>{
        'acceptanceId': acceptanceId == null ? 'n/a' : acceptanceId,
        'invoice': invoice,
        'invoiceNumber': invoiceNumber == null ? 'n/a' : invoiceNumber,
        'customer': customer,
        'user': user == null ? 'n/a' : user,
        'date': date == null ? 'n/a' : date,
        'supplierName': supplierName,
        'supplier': supplier,
        'customerName': customerName,
      };
}

File: HYPERLEDGER/bfn-network-chaincode/bfnwebapi-node/src/server/models/invoice-acceptance.ts

import { JsonObject, JsonProperty } from 'json2typescript';

@JsonObject('InvoiceAcceptance')
export class InvoiceAcceptance {
  @JsonProperty('acceptanceId', String)
  public acceptanceId: string | undefined;

  @JsonProperty('docType', String)
  public docType?: string;

  @JsonProperty('customerName', String)
  public customer: string | undefined;

  @JsonProperty('invoiceNumber', String)
  public invoiceNumber: string | undefined;
  @JsonProperty('supplier', String)
  public supplier: string | undefined;
  @JsonProperty('supplierName', String)
  public supplierName: string | undefined;
  @JsonProperty('invoice', String)
  public invoice: string | undefined;
  @JsonProperty('date', String)
  public date: string | undefined;

}
